### PR TITLE
move imagestream creation to more sensible spot

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@liggitt @stevekuznetsov ptal, this doesn't fix anything, but it makes it more obvious what the expected inputs to each new-app command are.
[test]

File: test/cmd/newapp.sh [hunk 1]
```diff
@@ -11,13 +11,12 @@ os::log::install_errexit
 
 # This test validates the new-app command
 
-os::cmd::expect_success 'oc create -f examples/image-streams/image-streams-centos7.json'
-
 os::cmd::expect_success_and_text 'oc new-app library/php mysql -o yaml' '3306'
 os::cmd::expect_failure 'oc new-app unknownhubimage -o yaml'
 # verify we can generate a Docker image based component "mongodb" directly
 os::cmd::expect_success_and_text 'oc new-app mongo -o yaml' 'library/mongo'
 # the local image repository takes precedence over the Docker Hub "mysql" image
```
Comment (liggitt) [inline test/cmd/newapp.sh]: is just checking for the text `mysql` in `os::cmd::expect_success_and_text 'oc new-app mysql -o yaml' 'mysql'` really testing precedence? I would expect it to contain `mysql` even if it was referencing dockerhub

Comment (bparees) [inline test/cmd/newapp.sh]: yeah good question, that check probably needs to be more stringent.

Comment (bparees) [inline test/cmd/newapp.sh]: updated, ptal.

File: test/cmd/newapp.sh [hunk 2]
```diff
@@ -11,22 +11,21 @@ os::log::install_errexit
...
+os::cmd::expect_success 'oc create -f examples/image-streams/image-streams-centos7.json'
 os::cmd::try_until_success 'oc get imagestreamtags mysql:latest'
 os::cmd::try_until_success 'oc get imagestreamtags mysql:5.5'
 os::cmd::try_until_success 'oc get imagestreamtags mysql:5.6'
-os::cmd::expect_success_and_text 'oc new-app mysql -o yaml' 'mysql'
+os::cmd::expect_success_and_not_text 'oc new-app mysql -o yaml' 'library/mysql'
```
Comment (liggitt) [inline test/cmd/newapp.sh]: when I run this locally with image streams imported, I get:
```
docker.io/centos/mysql-56-centos7@sha256:8476447ff6caa8e26045979f323c955db54d59d9b3029f387a5988572c8cbb84
```

Comment (liggitt) [inline test/cmd/newapp.sh]: are we wanting to check what the image change trigger is set to? if so, would this work instead:
```
os::cmd::expect_success_and_not_text 'oc new-app mysql --output-version=v1 --output="jsonpath={..imageChangeParams.from.*}"' 'ImageStreamTag.*mysql:latest'
```

Comment (bparees) [inline test/cmd/newapp.sh]: it's a not_text, i'm just making sure it did not pick up the library/mysql image, which it should not if the mysql imagestream is imported.

Comment (liggitt) [inline test/cmd/newapp.sh]: oh, duh

**Comment by bparees** [comment]
@liggitt any other comments?

**Comment by liggitt** [comment]
nope, LGTM

**Comment by bparees** [comment]
[merge]

**Pull request merged**